### ISSUE TITLE
Lag spike every few seconds

### ISSUE THREAD

**Issue by marcpro44** [Haggle1996/RevolutionPack#645]
The game freeze  every 45 seconds. I installed opis to see if I can find the problem. I noticed that a spike appear in the ping graph when the freeze occur.  I tried :
- Allocating more ram to minecraft (6gb)
- Disabling Enviromine
-  Creating a new world

And nothing changed. The only way I found to reduce the lag is to lower the chunks but the problem is still there. 

![freeze_opis](https://cloud.githubusercontent.com/assets/28878026/26331971/e83717a2-3f21-11e7-9b21-ced045aa4308.PNG)

**Comment by Haggle1996** [comment]
It's not enviromine. The messages in the log are there because the mod has more blocks to process, but it's throttled and can't process more blocks until the next cycle... This is a good thing.

Does the problem happen all the time, even when more or less standing still? Did you pregenerate your world? Try turning vsync off in your video settings.

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by marcpro44** [comment]
Yes, the problem happen all the time. The fact of moving does not seem to have an influence. But, it's very periodic and predictable (every exactly 40-45 seconds). 

I had the same problem with the first version of the pack. In the beginning is not very apparent. However, when you add machines, it gets worse and worse. I was getting  lag of about 3 seconds. At this time I decided to stop playing because it was unplayable. 

When something make à constant noise like machines, it's easy to hear that the sound stop for a few moment and then go very fast to pass the skipped time and return to normal. However, when it's happening, I can continue to move but the mobs, machines, chests, etc. doesn't react. In opis, the spikes only appear on the ping chart and not on the tick time chart.

I also tried disabling vsync and no change. For the world pregeneration, I don't know how to do it. I hear some people talking about that but never known exactly what it is.

**Comment by dieselfuelonly** [comment]
Late to the party, but I've seen my client do this if I'm using too much memory and Windows decides it's time to start using virtual memory.  Same behavior, very predictable when it comes to how often it occurs and the pause is 2-3 seconds.  After it started, even freeing up memory by closing whatever else I had running in the background wouldn't fix the problem until I restarted the client.  For what it's worth I have 12GB of RAM and 4GB allocated to Minecraft now.  As long as I don't leave something using a ton of RAM in the background now it's not an issue.
